txnsync/transactionCache.go

```diff
@@ -169,31 +198,100 @@ func (lru *transactionCache) acknowledge(seqs []uint64) {
 
 // add a given transaction ID to the short term cache.
 func (st *shortTermTransactionCache) add(txid transactions.Txid) {
-	if st.transactionsMap[txid] {
+	entry, exists := st.transactionsMap[txid]
+	if exists {
+		// promote
+		if entry.detach() {
```

Comment:
if `entry == st.first` we could just `st.first=entry.next` and not detach+insert it?

Reply:
sounds right but not sure if the code will become clearer.